File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.py
input()
m = list(map(int, input().split()))
input()
n = list(map(int, input().split()))

dic = {}
for i in m:
    dic[i] = dic.get(i, 0) + 1

print(" ".join([str(dic.get(i, 0)) for i in n]))

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.js
const fs = require('fs')

const [_, a, __, b] = fs.readFileSync('/dev/stdin').toString().trim().split("\n").map(c => c.trim())

const dic = {}

for(let i of a.split(" ")){
    dic[i] = (dic[i] ?? 0) + 1;
}
console.log(b.split(" ").map(c => dic[c] ?? 0).join(" "))
              